src/ArmTemplates/Extractor/EntityExtractors/PolicyExtractor.cs

```diff
@@ -38,6 +41,50 @@ public class PolicyExtractor : IPolicyExtractor
             this.templateBuilder = templateBuilder;
         }
 
+        public string GetCachedPolicyContent(PolicyTemplateResource policyTemplateResource, string baseFilesGenerationDirectory)
+        {
+            if (policyTemplateResource?.Properties is null)
+            {
+                this.logger.LogWarning("Policy was not initialized correctly {0}", policyTemplateResource.Name);
+                return string.Empty;
+            }
+
+            // easy code flow, if we already have a policy-content-file-full-path to use as a key for policy content caches
+            if (!string.IsNullOrEmpty(policyTemplateResource.Properties.PolicyContentFileFullPath))
+            {
+                if (this.policyPathContentCache.ContainsKey(policyTemplateResource.Properties.PolicyContentFileFullPath))
+                {
+                    return this.policyPathContentCache[policyTemplateResource.Properties.PolicyContentFileFullPath];
+                }
+                else
+                {
+                    this.logger.LogWarning("Policy content at '{0}' was initialized, but wasn't cached properly", policyTemplateResource.Properties.PolicyContentFileFullPath);
```

Comment:
I think we can meet with it - if we are dealing with reference types (and we do), than `Properties.PolicyContentFileFullPath` can be changed at some time (i.e. in unrefactored code) and it will make an impact here. So I am leaving log message for easier investigation